src/sidecar/logging.ts

```diff
@@ -130,3 +133,81 @@ export function appendSidecarLogToOutputChannel(line: string) {
       );
   }
 }
+
+/**
+ * Gather the sidecar's log lines and stderr lines into a single object.
+ * Parses the 20 most recent sidecar log lines into a structured format.
+ *
+ * @param sidecarLogfilePath The path to the sidecar log file.
+ * @param stderrPath The path to the sidecar's stderr file.
+ *
+ * @returns SidecarOutputs structure containing the log lines, parsed log lines, and stderr lines.
+ */
+export async function gatherSidecarOutputs(
+  sidecarLogfilePath: string,
+  stderrPath: string,
+): Promise<SidecarOutputs> {
+  const myLogger = logger.withCallpoint("gatherSidecarOutputs");
+  // Try to read+parse most recent 20 sidecar logs to notice any startup errors (occupied port, missing
+  // configs, etc.)
+  const reformattedLogLines: string[] = [];
+  const parsedLines: SidecarLogFormat[] = [];
+
+  let rawLogs: string[] = [];
+  try {
+    rawLogs = (await readFile(Uri.file(sidecarLogfilePath))).trim().split("\n").slice(-20);
+  } catch (e) {
+    myLogger.error(`Failed to read sidecar log file: ${e}`);
+  }
+
+  for (const rawLogLine of rawLogs) {
+    try {
+      const parsed = JSON.parse(rawLogLine.trim()) as SidecarLogFormat;
+      if (!parsed || !parsed.timestamp || !parsed.level || !parsed.loggerName || !parsed.message) {
+        throw new Error("Corrupted log line");
+      }
+      parsedLines.push(parsed);
+
+      const formatted = `\t> ${parsed.timestamp} ${parsed.level} [${parsed.loggerName}] ${parsed.message}`;
+      reformattedLogLines.push(formatted);
+    } catch {
+      // JSON parsing or post-JSON structure issue. Only append the raw line to logLines (if nonempty).
+      if (rawLogLine !== "") {
+        reformattedLogLines.push(rawLogLine);
+      }
+    }
+  }
+
+  let stderrLines: string[] = [];
+  try {
+    let stderrContent = await readFile(Uri.file(stderrPath));
+    stderrContent = stderrContent.trim();
+    stderrLines = stderrContent.split("\n");
+    if (stderrLines.length === 1 && stderrLines[0] === "") {
+      // File was essentialy empty. Coerce to empty array.
+      stderrLines = [];
+    }
+  } catch (e) {
+    myLogger.error(`Failed to read sidecar stderr file: ${e}`);
+  }
+
+  return {
+    logLines: reformattedLogLines,
+    parsedLogLines: parsedLines,
+    stderrLines: stderrLines,
+  };
+}
+
+export function divineSidecarStartupFailureReason(
```

Comment:
```suggestion
export function determineSidecarStartupFailureReason(
```
Minor rename suggestion